src/Home.jsx

```diff
@@ -1,69 +1,51 @@
 import React, { useState } from 'react';
-import { useLoaderData } from 'react-router-dom'
+import { useLoaderData } from 'react-router-dom';
 
 import CharacterList from './components/characterList';
 import ScriptNav from './components/ScriptNav';
 import Upload from './components/Upload';
 import ActorList from './components/ActorList';
 import AddActor from './components/AddActor';
-import Header from './components/Header'
+import Header from './components/Header';
+import { getActors, getScriptTitles } from './components/api';
 
 
 // Load the actor names and script titles on page load
 export async function loader() {
-  const [loadedActors, loadedScripts] = await Promise.all([getActors(), getScriptTitles()])
-  return { loadedActors, loadedScripts}
-}
-
-//TODO: Consolidate api call functions into one file OR extract out common functionality for fetch/handle error
-
-async function getActors(){
-  try {
-    const response = await fetch('/api/actors')
-
-    if (response.ok) {
-      return await response.json()
-    } else {
-      console.log('Server Error:', response.body);
-    }
-  } catch (error) {
-    console.error('Network Error:', error);
-  }
-}
-
-async function getScriptTitles(){
-  try {
-    const response = await fetch('/api/scripts/title');
-
-    if (response.ok) {
-      return await response.json();
+  const [loadedActors, loadedScripts] = await Promise.all([getActors(), getScriptTitles()]);
+  if (loadedActors.success && loadedScripts.success) {
+    return { loadedActors: loadedActors.data, loadedScripts: loadedScripts.data };
+  } else {
+    if (loadedActors.success === false) {
```

Comment:
nit: this syntax can be simplified slightly
```js
if (a) {

} else {
    if (b) {

    } else {

    }
}
```
is the same as this (which has less nesting)
```js
if (a) {

} else if (b) {

} else {

}
```

Reply:
Right, much simpler syntax. This ended up being removed but I appreciate the reminder!